plugins/libplugin.h

```diff
@@ -50,6 +50,14 @@ struct plugin_notification {
 	               const jsmntok_t *params);
 };
 
+/* Create an array of these, one for each hook you subscribe to. */
+struct plugin_hook {
+	const char *name;
+	struct command_result (*handle)(struct command *cmd,
+	                                const char *buf,
+	                                const jsmntok_t *params);
+};
+
```

Comment:
LGTM tested notifications and hooks. :+1: 
I had to change  

> struct command_result (*handle)(struct command *cmd,
	                                const char *buf,
	                                const jsmntok_t *params);
}

to a function pointer to command_result

To get the hook to return a command_success(cmd, ...);


```
diff --git a/plugins/libplugin.h b/plugins/libplugin.h
index d79e7e53..b1df9cd7 100644
--- a/plugins/libplugin.h
+++ b/plugins/libplugin.h
@@ -53,7 +53,7 @@ struct plugin_notification {
 /* Create an array of these, one for each hook you subscribe to. */
 struct plugin_hook {
        const char *name;
-       struct command_result (*handle)(struct command *cmd,
+       struct command_result *(*handle)(struct command *cmd,
                                        const char *buf,
                                        const jsmntok_t *params);
 };

```

Reply:
Oops, thanks !